### PR TITLE
Overloading OpenConnection function

### PR DESCRIPTION
**Pull request by rodrigomurta**
Using Connection with Credential to connect using user and password. Tested in Mongo 3.2

### CODE REVIEW

File: MongoDBLink.m [hunk 1]
```diff
@@ -124,6 +126,18 @@
   DatabaseConnection[host, port, JavaNew["com.mongodb.MongoClient", host, port]]
 ]
 
+OpenConnection[host_String, port_Integer, "", _, _]:= OpenConnection[host, port]
```
Comment (zbjornson) [inline MongoDBLink.m:129]: What's the use case for this overload? Having a blank string seems to make it very specific, or is that standard for indicating anonymous authentication?

File: MongoDBLink.m [hunk 2]
```diff
@@ -124,6 +126,18 @@
...
+
+OpenConnection[host_String, port_Integer, user_String, password_String, database_String] := Module[{al, credential},
+    InstallJava[];
+	LoadJavaClass["com.mongodb.MongoCredential"];
```
Comment (zbjornson) [inline MongoDBLink.m:133]: Minor: inconsistent indentation in the Module body.

**Comment by zbjornson** [comment]
Thanks for the PR!

File: MongoDBLink.m (hunk 1 above)
Comment (rodrigomurta) [inline MongoDBLink.m:129]: @zbjornson you got the point. When you want anonymous authentication, you can still use OpenConnection with 5 parameters, using user as "". In my case, I read these parameters from a txt file, and I can still read the same number of fields to use OpenConnection, even for anonymous case. Using user -> "", pass -> "" , base -> "". And what matter is the user.

**Comment by rodrigomurta** [comment]
@zbjornson Thank you for MongoDBLink, It's a very useful package. I use a lot.

**Pull request merged**